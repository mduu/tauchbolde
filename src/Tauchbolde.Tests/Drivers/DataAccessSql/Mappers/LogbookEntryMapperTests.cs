using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using Newtonsoft.Json;
using Tauchbolde.Driver.DataAccessSql.DataEntities;
using Tauchbolde.Driver.DataAccessSql.Mappers;
using Xunit;

namespace Tauchbolde.Tests.Drivers.DataAccessSql.Mappers
{
    [UseReporter(typeof(DiffReporter))]
    public class LogbookEntryMapperTests
    {
        [Fact]
        public void Map()
        {
            var dataObject = new LogbookEntryData
            {
                Id = new Guid("08F81971-81C9-4ED8-932D-95940B6E27A3"),
                Title = "The Title",
                TeaserText = "The Teaser",
                Text = "The Text",
                CreatedAt = new DateTime(2019, 8, 1, 12, 0, 0),
                ModifiedAt = new DateTime(2019, 8, 2, 13, 0, 0),
                IsFavorite = true,
                IsPublished = true,
                EventId = new Guid("6D3CD14C-15AF-4BAE-8CC1-A0B6A9EE397D"),
                TeaserImage = "TheTeaserImage",
                TeaserImageThumb = "TheTeaserImageThumb",
                OriginalAuthorId = new Guid("BFBEA88B-3A40-4487-9FC4-B477DD20E17C"),
                EditorAuthorId = new Guid("5A835146-69F7-4A19-BE1B-E7501D01BD8D"),
                ExternalPhotoAlbumUrl = "https://tauchbolde.ch",
            };

            var domainObject = dataObject.MapTo();
            var newDataObject = domainObject.MapTo();
            
            newDataObject.Should().BeEquivalentTo(dataObject);
            Approvals.VerifyJson(JsonConvert.SerializeObject(new
            {
                dataObject,
                domainObject,
                newDataObject
            }));
        }
    }
}