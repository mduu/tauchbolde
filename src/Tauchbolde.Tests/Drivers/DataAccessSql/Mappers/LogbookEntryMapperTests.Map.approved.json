{
    "dataObject": {
        "Title": "The Title",
        "Text": "The Text",
        "TeaserText": "The Teaser",
        "IsFavorite": true,
        "TeaserImage": "TheTeaserImage",
        "TeaserImageThumb": "TheTeaserImageThumb",
        "ExternalPhotoAlbumUrl": "https://tauchbolde.ch",
        "CreatedAt": "2019-08-01T12:00:00",
        "ModifiedAt": "2019-08-02T13:00:00",
        "IsPublished": true,
        "EditorAuthorId": "5a835146-69f7-4a19-be1b-e7501d01bd8d",
        "EditorAuthor": null,
        "OriginalAuthorId": "bfbea88b-3a40-4487-9fc4-b477dd20e17c",
        "OriginalAuthor": null,
        "EventId": "6d3cd14c-15af-4bae-8cc1-a0b6a9ee397d",
        "Event": null,
        "Id": "08f81971-81c9-4ed8-932d-95940b6e27a3"
    },
    "domainObject": {
        "Title": "The Title",
        "Text": "The Text",
        "TeaserText": "The Teaser",
        "IsFavorite": true,
        "TeaserImage": "TheTeaserImage",
        "TeaserImageThumb": "TheTeaserImageThumb",
        "ExternalPhotoAlbumUrl": "https://tauchbolde.ch",
        "CreatedAt": "2019-08-01T12:00:00",
        "ModifiedAt": "2019-08-02T13:00:00",
        "IsPublished": true,
        "EditorAuthorId": "5a835146-69f7-4a19-be1b-e7501d01bd8d",
        "EditorAuthor": null,
        "OriginalAuthorId": "bfbea88b-3a40-4487-9fc4-b477dd20e17c",
        "OriginalAuthor": null,
        "EventId": "6d3cd14c-15af-4bae-8cc1-a0b6a9ee397d",
        "Event": null,
        "Id": "08f81971-81c9-4ed8-932d-95940b6e27a3"
    },
    "newDataObject": {
        "Title": "The Title",
        "Text": "The Text",
        "TeaserText": "The Teaser",
        "IsFavorite": true,
        "TeaserImage": "TheTeaserImage",
        "TeaserImageThumb": "TheTeaserImageThumb",
        "ExternalPhotoAlbumUrl": "https://tauchbolde.ch",
        "CreatedAt": "2019-08-01T12:00:00",
        "ModifiedAt": "2019-08-02T13:00:00",
        "IsPublished": true,
        "EditorAuthorId": "5a835146-69f7-4a19-be1b-e7501d01bd8d",
        "EditorAuthor": null,
        "OriginalAuthorId": "bfbea88b-3a40-4487-9fc4-b477dd20e17c",
        "OriginalAuthor": null,
        "EventId": "6d3cd14c-15af-4bae-8cc1-a0b6a9ee397d",
        "Event": null,
        "Id": "08f81971-81c9-4ed8-932d-95940b6e27a3"
    }
}